File: ApiProject/API-TFG/src/main/java/com/aitor/api_tfg/model/request/FiltroRequest.java
package com.aitor.api_tfg.model.request;

import com.aitor.api_tfg.model.db.Modalidades;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class FiltroRequest {
    private String nombre;
    private Modalidades activityType;

    private float distanciaMin;
    private float distanciaMax;

    private double positiveElevationMin;
    private double positiveElevationMax;

    private long durationMin;
    private long durationMax;

    private float averageSpeedMin;
    private float averageSpeedMax;
}

File: AndroidProject/app/src/main/java/com/aitor/trackactividades/historialActividades/data/request/FiltroRequest.kt
package com.aitor.trackactividades.historialActividades.data.request

import com.aitor.trackactividades.core.model.Modalidades

data class FiltroRequest(
    var nombre: String? = null,
    var activityType: Modalidades? = null,

    var distanciaMin: Float = 0f,
    var distanciaMax: Float = 0f,

    var positiveElevationMin: Double = 0.0,
    var positiveElevationMax: Double = 0.0,

    var durationMin: Long = 0L,
    var durationMax: Long = 0L,

    var averageSpeedMin: Float = 0f,
    var averageSpeedMax: Float = 0f
)
